### ISSUE TITLE
RF02 falsely triggering for SELECT * EXCEPT (...) in Databricks

### ISSUE THREAD

**Issue by TimWitzel** [sqlfluff/sqlfluff#5764]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

For statements in databricks SQL which use "SELECT table.* EXCEPT (column_name)" RF02 is triggered even though it probably should not be, since the reference to the table is clear.

![image](https://github.com/sqlfluff/sqlfluff/assets/61972354/e8b97f53-7e6f-47e5-aede-238237f1103c)
![image](https://github.com/sqlfluff/sqlfluff/assets/61972354/065ff16e-dc30-46e0-9510-85b30c871997)

### Expected Behaviour

RF02 should not trigger if the unqualified Reference is part of the EXCEPT clause in the SELECT * context.

### Observed Behaviour

![image](https://github.com/sqlfluff/sqlfluff/assets/61972354/92c2a8f8-d99d-431c-8ec6-1c7ae15ac0f9)
RF02 triggers.

### How to reproduce

Minimal SQL: 
```
SELECT 
    test.* EXCEPT (test_column),
    test_2.*
FROM test
    LEFT JOIN test_2 ON test.test_column = test_2.test_column
```
`
sqlfluff lint test.sql --dialect databricks`

### Dialect

Databricks

### Version

3.0.3

### Configuration

```
[sqlfluff]
dialect = databricks
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by data-blade** [comment]
+1

apart from `except()` this rule should also ignore unqualified columns if the only join type is `semi`/`anti` (both supported by databricks)
